Applications/Spire/Source/Ui/DecimalInputWidget.cpp

```diff
@@ -46,10 +46,10 @@ DecimalInputWidget::DecimalInputWidget(double value, QWidget* parent)
 }
 
 QString DecimalInputWidget::textFromValue(double value) const {
-  if(value == 0.0) {
+  if(value == 0) {
```

Comment:
0 is an integer literal. When comparing a value against a literal, always try to supply the same type of the literal as that of the value.
"0." (zero period) is a double literal and is a short form of "0.0".

Reply:
If the output is the same I'd prefer to keep the trailing zero to make it clearer.

Reply:
Okay, restore the original double literal. If trailing zeros are prohibited (I do not know if they are), Kamal will probably let you know.